jscatter/encodings.py

```diff
@@ -5,14 +5,44 @@
 from math import floor
 from typing import List, Tuple, Union
 
-def create_legend(values, encoding, norm, categories, linspace_num=5):
+def create_legend(encoding, norm, categories, labeling=None, linspace_num=5):
+    variable = None
+    if labeling:
+        try:
+            variable = labeling['variable']
+        except KeyError as e:
+            variable = None
+    values = []
+
     if categories:
         assert len(categories) == len(encoding), 'The categories and encoding need to be of the same size'
         cat_by_idx = { catIdx: cat for cat, catIdx in categories.items() }
-        return [(cat_by_idx[i], encoding[i]) for i in range(len(cat_by_idx))]
-
-    S = np.linspace(0, 1, linspace_num)
-    return [(norm.inverse(s), encoding[floor((len(encoding) - 1) * s)]) for s in S]
+        values = [(cat_by_idx[i], encoding[i], None) for i in range(len(cat_by_idx))]
+    else:
+        S = np.linspace(0, 1, linspace_num)
+        for s in S:
+            label = None
+
+            if labeling:
+                if s == 0:
+                    try:
+                        label = labeling['minValue']
+                    except KeyError as e:
+                        label = None
+
+                if s == S[-1]:
+                    try:
+                        label = labeling['maxValue']
+                    except KeyError as e:
+                        label = None
```

Comment:
I wonder if this would be easier to read if we just created the `values` upfront with all `None` labels, and then modified the `values[0]` and `values[-1]` entries if `labeling` and `minValue`/`maxValue` are provided.

Reply:
I was thinking about that but then how would we quickly judge if the user ever provided any labeling? And how would we elegantly handle cases where a user wants to unset the labeling, e.g., `scatter.color(labeling=None)`?

Reply:
I guess my thought was something like,

```python
values = [
    (norm.inverse(s), encoding[floor((len(encoding) - 1) * s)], None)
    for s in np.linspace(0, 1, linspace_num)
]

if labeling:
    tmp = values[0]
    values[0] = (tmp[0], tmp[1], labeling.get('minValue', tmp[2]))

    tmp = values[-1]
    values[-1] = (tmp[0], tmp[1], labeling.get('maxValue', tmp[2]))

```

if `labeling` is explicitly `None`, then all values have `None` for a label. I'm probably missing something...

Reply:
Got it! I've optimized it a bit further for conciseness. What do you think?